src/validation.cpp

```diff
@@ -3066,6 +3086,19 @@ static bool AcceptBlockHeader(const CBlockHeader& block, CValidationState& state
             return state.DoS(100, error("%s: prev block invalid", __func__), REJECT_INVALID, "bad-prevblk");
         if (!ContextualCheckBlockHeader(block, state, chainparams, pindexPrev, GetAdjustedTime()))
             return error("%s: Consensus::ContextualCheckBlockHeader: %s, %s", __func__, hash.ToString(), FormatStateMessage(state));
+
+        for (const CBlockIndex* failedit : g_failed_blocks) {
+            if (pindexPrev->GetAncestor(failedit->nHeight) == failedit) {
+                assert(failedit->nStatus & BLOCK_FAILED_VALID);
+                CBlockIndex* invalid_walk = pindexPrev;
+                while (invalid_walk != failedit) {
+                    invalid_walk->nStatus |= BLOCK_FAILED_CHILD;
+                    setDirtyBlockIndex.insert(invalid_walk);
+                    invalid_walk = invalid_walk->pprev;
+                }
+                return state.DoS(100, error("%s: prev block invalid", __func__), REJECT_INVALID, "bad-prevblk");
```

Comment:
It would be nice to specifically test for `bad-prevblk` being thrown, perhaps using the mechanism in #11220. I'm not sure how to do that in the python tests though.

Reply:
Another place to test this would be to add a test for [ProcessNewBlockHeaders](https://github.com/bitcoin/bitcoin/blob/v0.15.0.1/src/validation.cpp#L3067-L3084). I can take a stab at it, but I'm not sure which file that test should be added too, or which test would be a good example to base it on.